infra/outlines.py

```diff
@@ -93,21 +93,77 @@ def write_html(objs, indent=0):
             write_html(subs, indent=indent+4)
         print("</code>")
 
+def is_doc_string(cmd):
+    return isinstance(cmd, ast.Expr) and \
+        isinstance(cmd.value, ast.Constant) and isinstance(cmd.value.value, str)
+
+def is_sys_modules_hack(cmd):
+    return isinstance(cmd, ast.Assign) and len(cmd.targets) == 1 and \
+        isinstance(cmd.targets[0], ast.Attribute) and \
+        isinstance(cmd.targets[0].value, ast.Subscript) and \
+        isinstance(cmd.targets[0].value.value, ast.Attribute) and \
+        isinstance(cmd.targets[0].value.value.value, ast.Name) and \
+        cmd.targets[0].value.value.value.id == 'sys' and \
+        cmd.targets[0].value.value.attr == 'modules'
+
+def is_if_main(cmd):
+    return isinstance(cmd, ast.If) and isinstance(cmd.test, ast.Compare) and \
+        isinstance(cmd.test.left, ast.Name) and cmd.test.left.id == "__name__" and \
+        len(cmd.test.comparators) == 1 and isinstance(cmd.test.comparators[0], ast.Constant) and \
+        cmd.test.comparators[0].value == "__main__" and len(cmd.test.ops) == 1 and \
+        isinstance(cmd.test.ops[0], ast.Eq)
+
+def get_name(module, name):
+    assert isinstance(module, ast.Module)
+    for item in module.body:
+        if isinstance(item, ast.Import): pass
```

Comment:
Is there a way to share code so that you don't have to repeat the calls like is_sys_modules_hack here and in to_item?

Reply:
I think avoiding the copy-and-paste here would require more boilerplate (making a custom Visitor?) than it would save.